src/overlays/actors/ovl_Obj_Jgame_Light/z_obj_jgame_light.c

```diff
@@ -28,29 +33,173 @@ const ActorInit Obj_Jgame_Light_InitVars = {
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON | BUMP_HOOKABLE,
+        OCELEM_ON,
+    },
     { 12, 45, 0, { 0, 0, 0 } },
 };
 
-#endif
-
-extern ColliderCylinderInit D_80C15BC0;
-
-extern UNK_TYPE D_060003A0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/ObjJgameLight_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/ObjJgameLight_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/func_80C15474.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/func_80C156C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/func_80C15718.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/ObjJgameLight_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/ObjJgameLight_Draw.s")
+#include "assets/overlays/ovl_Obj_Jgame_Light/ovl_Obj_Jgame_Light.c"
+
+void ObjJgameLight_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjJgameLight* this = THIS;
+    LightInfo* lights = &this->lightInfo;
+
+    Actor_SetScale(&this->actor, 1.0f);
+    func_800B4AEC(globalCtx, &this->actor, 50.0f);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B4B50, 1.0f);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    Lights_PointGlowSetInfo(lights, this->actor.world.pos.x, (this->actor.world.pos.y + 70.0f), this->actor.world.pos.z,
+                            0xFF, 0xFF, 0xB4, -1);
+    this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+    Actor_SetFocus(&this->actor, 60.0f);
+    this->actor.colChkInfo.health = 0;
+    this->unk_1B8 = 0;
+    this->unk_1B6 = 0;
+    this->unk_1AC = 0;
+    this->unk_1B2 = 0;
+    this->unk_1B4 = 0;
+    this->unk_1A8 = 0.0f;
+}
+
+void ObjJgameLight_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjJgameLight* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+}
+
+void func_80C15474(ObjJgameLight* this, GlobalContext* globalCtx) {
+    u8 temp_a1;
+
+    if (((this->actor.colChkInfo.health & 1) != 0) && (this->unk_1B6 == 0)) {
+        if (this->unk_1AC < 0xA0) {
+            this->unk_1AC += 0x28;
+        } else {
+            this->unk_1AC = 0xC8;
+            this->unk_1B6 = 1;
+        }
+        if (this->unk_1A8 < 0.7f) {
+            this->unk_1A8 += 0.3f;
+        } else {
+            this->unk_1A8 = 1.0f;
+        }
+    } else if ((this->actor.colChkInfo.health & 8) != 0) {
+        if (this->unk_1AC >= 0x29) {
+            this->unk_1AC -= 0x28;
+        } else {
+            this->unk_1AC = -1;
+            if (this->unk_1A8 == 0.0f) {
+                this->unk_1B6 = 0;
+                this->actor.colChkInfo.health &= 0xFFFE;
+                this->actor.colChkInfo.health &= (s16)0xFFF7;
```

Comment:
```suggestion
                this->actor.colChkInfo.health &= ~2;
                this->actor.colChkInfo.health &= ~8;
```
if I remember correctly.